### PR TITLE
jenkins update and security fix

### PR DESCRIPTION
**Pull request by chris530**

### CODE REVIEW

**Comment by wgrzelak** [comment]
It seems like Jenkins 2.59 is the weekly releases (latest), Jenkins 2.46.2 is LTS (stable).

What do you think to support both, latest as 2.5.9 and stable as 2.46.2?

**Comment by huyhg** [comment]
Seconding Wojciech, although I think we only need the LTS version, which is 2.46.

Chris, do you know if 2.46 resolves the security issue you referenced?

**Comment by chris530** [comment]
SECURITY-429 / CVE-2017-1000353
https://jenkins.io/changelog-stable/

Looks like 2.46.2  ( stable ) fixes it

**Comment by chris530** [comment]
PR updated to 2.46.2

**Comment by chris530** [comment]
Successfully launched with docker-compose, should be good to go.

File: versions.yaml [hunk 1]
```diff
@@ -18,17 +18,16 @@
 # CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
 
 versions:
-  - dir: '2/2.32'
+  - dir: '2/2.46.2'
     repo: jenkins2
     tags:
-      - 2.32
-      - 2.32.2
+      - 2.46.2
```
Comment (huyhg) [inline versions.yaml]: can we also add 2.46 tag?

File: 2/2.46.2/Dockerfile [hunk 2]
```diff
@@ -38,10 +38,10 @@ RUN mkdir -p /usr/share/doc/tini && curl -fsSL https://raw.githubusercontent.com
 COPY init.groovy /usr/share/jenkins/ref/init.groovy.d/tcp-slave-agent-port.groovy
 
 # jenkins version being bundled in this docker image
-ENV JENKINS_VERSION 2.32.2
+ENV JENKINS_VERSION 2.46.2
```
Comment (huyhg) [inline 2/2.46.2/Dockerfile]: hmm, should we just use `2/` directory? I don't think renaming the directory for every update is very covenient

**Comment by chris530** [comment]
Also added 2.46 tag

tags are now

latest
2.46
2.46.2

File: 2/2.46.2/Dockerfile (hunk 2 above)
Comment (chris530) [inline 2/2.46.2/Dockerfile]: Assuming there is no major security issue,  do we plan to leave old versions ( and tags ) available to the users ?

File: versions.yaml (hunk 1 above)
Comment (wgrzelak) [inline versions.yaml]: The stable branch and latest branch have different versions, it means that `2.46` was never available in stable branch, but in latest it was.

https://jenkins.io/changelog/
https://jenkins.io/changelog-stable/

Maybe should we keep `2`, `2.46.2` and `latest` in this case?

File: 2/2.46.2/Dockerfile (hunk 2 above)
Comment (wgrzelak) [inline 2/2.46.2/Dockerfile]: +1

File: versions.yaml (hunk 1 above)
Comment (huyhg) [inline versions.yaml]: We should remove the 2.32.x track. So Jenkins now only has one latest track. I'll file a bug to track this.

Comment (huyhg) [inline versions.yaml]: Didn't address Wojciech's comment: We should also add tag `2` to this version for completeness as well, even though Launcher doesn't use tags other than `latest`.

**Pull request merged**